crates/router/src/connector/square/transformers.rs

```diff
@@ -1,95 +1,268 @@
+}
+#[derive(Debug, Serialize)]
+pub struct SquarePaymentsRequestExternalDetails {
+    source: String,
+    #[serde(rename = "type")]
+    source_type: String,
+}
+
+#[derive(Debug, Serialize)]
+pub struct SquarePaymentsRequest {
+    amount_money: SquarePaymentsAmountData,
+    idempotency_key: Secret<String>,
+    source_id: Secret<String>,
+    autocomplete: bool,
+    external_details: SquarePaymentsRequestExternalDetails,
 }
 
 impl TryFrom<&types::PaymentsAuthorizeRouterData> for SquarePaymentsRequest {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
+        let autocomplete = item.request.is_auto_capture()?;
         match item.request.payment_method_data.clone() {
-            api::PaymentMethodData::Card(req_card) => {
-                let card = SquareCard {
-                    name: req_card.card_holder_name,
-                    number: req_card.card_number,
-                    expiry_month: req_card.card_exp_month,
-                    expiry_year: req_card.card_exp_year,
-                    cvc: req_card.card_cvc,
-                    complete: item.request.is_auto_capture()?,
-                };
-                Ok(Self {
+            api::PaymentMethodData::Card(_) => Ok(Self {
+                idempotency_key: Secret::new(item.payment_id.clone()),
+                source_id: Secret::new(item.get_payment_method_token()?),
+                amount_money: SquarePaymentsAmountData {
                     amount: item.request.amount,
-                    card,
-                })
-            }
-            _ => Err(errors::ConnectorError::NotImplemented("Payment methods".to_string()).into()),
+                    currency: item.request.currency,
+                },
+                autocomplete,
+                external_details: SquarePaymentsRequestExternalDetails {
+                    source: "Hyperswitch".to_string(),
+                    source_type: "Card".to_string(),
+                },
+            }),
+            api::PaymentMethodData::BankDebit(_)
+            | api::PaymentMethodData::CardRedirect(_)
+            | api::PaymentMethodData::Wallet(_)
+            | api::PaymentMethodData::PayLater(_)
+            | api::PaymentMethodData::MandatePayment
+            | api::PaymentMethodData::GiftCard(_)
+            | api::PaymentMethodData::Upi(_) => Err(errors::ConnectorError::NotImplemented(
+                "Payment Method".to_string(),
+            ))
+            .into_report(),
+            api::PaymentMethodData::BankRedirect(_)
+            | api::PaymentMethodData::BankTransfer(_)
+            | api::PaymentMethodData::Crypto(_)
+            | api::PaymentMethodData::Reward(_)
+            | api::PaymentMethodData::Voucher(_) => Err(errors::ConnectorError::NotSupported {
+                message: format!("{:?}", item.request.payment_method_data),
+                connector: "Square",
+            })?,
```

Comment:
Same here, Let's have different try_from for each payment method data

Reply:
This is not required because every payment method will require the same function because of tokenization flow.